frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/ConnectionSection.tsx

```diff
@@ -0,0 +1,310 @@
+    </>
+  );
+};
+
+type NewConnectionFieldProps = {
+  connectionTypes: ConnectionTypeConfigMapObj[];
+  data: CreatingInferenceServiceObject;
+  setData: UpdateObjectAtPropAndValue<CreatingInferenceServiceObject>;
+  setNewConnection: (connection: Connection) => void;
+  setIsConnectionvalid: (isValid: boolean) => void;
+};
+
+const NewConnectionField: React.FC<NewConnectionFieldProps> = ({
+  connectionTypes,
+  data,
+  setData,
+  setNewConnection,
+  setIsConnectionvalid,
+}) => {
+  const enabledConnectionTypes = React.useMemo(
+    () =>
+      connectionTypes.filter((t) => t.metadata.annotations?.['opendatahub.io/enabled'] === 'true'),
+    [connectionTypes],
+  );
+  const [selectedConnectionType, setSelectedConnectionType] = React.useState<
+    ConnectionTypeConfigMapObj | undefined
+  >(
+    enabledConnectionTypes.length === 1
+      ? withRequiredFields(connectionTypes[0], [
+          'AWS_ACCESS_KEY_ID',
+          'AWS_SECRET_ACCESS_KEY',
+          'AWS_S3_ENDPOINT',
+          'AWS_S3_BUCKET',
+        ])
+      : undefined,
+  );
+  const { data: nameDescData, onDataChange: setNameDescData } = useK8sNameDescriptionFieldData();
+
+  const [connectionValues, setConnectionValues] = React.useState<{
+    [key: string]: ConnectionTypeValueType;
+  }>(enabledConnectionTypes.length === 1 ? getDefaultValues(enabledConnectionTypes[0]) : {});
+
+  const [validations, setValidations] = React.useState<{
+    [key: string]: boolean;
+  }>({});
+  const isFormValid = React.useMemo(
+    () =>
+      !!selectedConnectionType &&
+      !!nameDescData.name &&
+      !selectedConnectionType.data?.fields?.find(
+        (field) =>
+          isConnectionTypeDataField(field) &&
+          field.required &&
+          !connectionValues[field.envVar] &&
+          field.type !== ConnectionTypeFieldType.Boolean,
+      ) &&
+      !Object.values(validations).includes(false),
+    [selectedConnectionType, nameDescData, connectionValues, validations],
+  );
+
+  React.useEffect(() => {
+    if (selectedConnectionType) {
+      setNewConnection(
+        assembleConnectionSecret(
+          data.project,
+          getResourceNameFromK8sResource(selectedConnectionType),
+          nameDescData,
+          connectionValues,
+        ),
+      );
+      setIsConnectionvalid(isFormValid);
+    }
+  }, [
+    connectionValues,
+    data.project,
+    isFormValid,
+    nameDescData,
+    selectedConnectionType,
+    setIsConnectionvalid,
+    setNewConnection,
+  ]);
+
+  return (
+    <FormSection>
+      <ConnectionTypeForm
+        options={enabledConnectionTypes}
+        connectionType={selectedConnectionType}
+        setConnectionType={(type) => {
+          setSelectedConnectionType(
+            withRequiredFields(
+              connectionTypes.find((t) => getResourceNameFromK8sResource(t) === type),
+              ['AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY', 'AWS_S3_ENDPOINT', 'AWS_S3_BUCKET'],
```

Comment:
Should be a const set of env vars that is shared with `isModelServingCompatibility`.